students/dave_arasim/session04/mailroom2.py

```diff
@@ -0,0 +1,222 @@
+        except ValueError:
+            user_choice = ''
+
+        if user_choice == '':
+            print('Sorry, but that is not a number')
+        else:
+            try:
+                switch_dict.get(user_choice)(donor_dict)
+            except TypeError:
+                print('That option: ', user_choice, ' is out of range', sep='')
+
+        print()
+
+        if quit_option:
+            print('Quitting process...')
+        else:
+            print('<cr> to continue... ', end='')
+            cont_input = input()  #cont_input for pause (variable not used)
+
+    build_donor_text(donor_dict)  #Export the Donor Dictionary back to a text file
+
+def quit_option_true(donor_dict):
+    '''Set the quit_option to true, so function can be selected from dict switch'''
+    global quit_option
+    quit_option = True
+
+def build_donor_dict(donor_dict):
+    '''Build Donor Dictionary by extracting values from a text file'''
+    try:
+        with open('donor_dict.txt', 'r') as dictfile:
+            while True:
+                donor_dict_line = dictfile.readline()
+                if not donor_dict_line: break
+
+                donor_dict_line = donor_dict_line.strip()
+                donor_dict_name = donor_dict_line.split(':')[0]
+                donor_dict_amts = donor_dict_line.split(':')[1].split(',')
+
+                for amts in donor_dict_amts:
+                    donor_dict[donor_dict_name].append(float(amts))
+
+        dictfile.close()
+    except FileNotFoundError:
+        print()
+        print('Donor database in donor_dict.txt not found.  Starting with new, empty database.')
+        print()
+
+def build_donor_text(donor_dict):
+    '''Export the Donor Dictionary date back to a text file'''
+    with open('donor_dict.txt', 'w') as dictfile:
+        for this_name, this_amts in donor_dict.items():
+            amts_str  = ','.join(str(x) for x in this_amts)
+            this_line = this_name + ':' + amts_str + '\n'
+
+            dictfile.write(this_line)
+
+    dictfile.close()
+
+def send_thank_you(donor_dict):
+    '''Sends thank you letter to single donor as their donation is made and recorded'''
+    global this_os
+    quit_thank_you = False
+
+    if this_os == 'Windows':
+        os.system('clear')
+    else:
+        os.system('cls')
+
+    while not(quit_thank_you):
+        print("Enter Donor's name, type 'list' for existing Donors (<cr> to exit): ", end='')
+
+        thank_you_choice = input()
+        print()
+
+        if   thank_you_choice == '':     quit_thank_you = True
+        elif thank_you_choice == 'list': show_donor_list(donor_dict)
+        else:
+            find_donor(donor_dict, thank_you_choice)
+            quit_thank_you = True
+
+def show_donor_list(donor_dict):
+    '''Show the Donor List of names only'''
+    donor_list   = 'List of Donors'
+    donor_list_w = 26
+
+    #Donor list header
+    list_nom = (donor_list, donor_list_w)
+    print('{:{}}'.format(*list_nom))
```

Comment:
This is still cryptic to me... Never knew braces can be nested in `.format()`!!! Learning new things every time!